src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -562,9 +564,28 @@ private bool IsUnsupportedType(ITypeSymbol type)
                     return true;
                 }
 
+                if (rootCall)
+                {
+                    ts_visitedTypes ??= new HashSet<ITypeSymbol>(SymbolEqualityComparer.Default);
+                    ts_visitedTypes.Clear();
```

Comment:
It's odd for this to be thread-static state, then the clearing behavior to be controlled by a parameter only specified by this function.  Could you instead just pass this as the optional parameter, and create it when the value is null?